controllers/drplacementcontrol_controller.go

```diff
@@ -1591,16 +1580,50 @@ func (r *DRPlacementControlReconciler) updateResourceCondition(
 		}
 
 		drpc.Status.ResourceConditions.ResourceMeta.ProtectedPVCs = protectedPVCs
-		drpc.Status.LastGroupSyncTime = vrg.Status.LastGroupSyncTime
-		drpc.Status.LastGroupSyncDuration = vrg.Status.LastGroupSyncDuration
-		drpc.Status.LastGroupSyncBytes = vrg.Status.LastGroupSyncBytes
-
-		if drpcMetrics != nil {
-			r.setLastSyncTimeMetric(&drpcMetrics.SyncMetrics, vrg.Status.LastGroupSyncTime, log)
-			r.setLastSyncDurationMetric(&drpcMetrics.SyncDurationMetrics, vrg.Status.LastGroupSyncDuration, log)
-			r.setLastSyncBytesMetric(&drpcMetrics.SyncDataBytesMetrics, vrg.Status.LastGroupSyncBytes, log)
+
+		if vrg.Status.LastGroupSyncTime != nil || drpc.Spec.Action != rmn.ActionRelocate {
```

Comment:
Why are we checking if the action is not relocate?

We should set the `drpc.Status.LastGroupSyncTime` to whatever value `vrg.Status.LastGroupSyncTime` contains as long as `GetVRGFromManagedCluster()` returns a valid VRG.
Before, we used to reset `drpc.Status.LastGroupSyncTime` to 0 even when `GetVRGFromManagedCluster()` failed to retrieve a valid VRG.  So what I am suggesting here, is to remove line 1584 entirely.

Reply:
wanted to handle the `0` value case even during the relocation.  So we do not want to update the DRPC LastGroup status during relocation. Wanted to use the last known good value. @ShyamsundarR please correct me if I am wrong here.

Reply:
@BenamarMk 

The issue here is, post relocate initially the VRG (VR and rbd image meta) would reflect a "0" last sync time. This is till the first reverse sync is performed which is expected to be shorter as it is a delta sync.

In this interval we retain the last known sync time for any informed failover action that may be performed in this state. For this if the vrg reported sync time is nil, we do not update the drpc sync time if the action noted was Relocate.

For failover vrg would again report a nil sync time till the initial sync is complete, but this would be reflected and updated in drpc as "0" and would raise the alerts. Similarly post initial deployment the sync time would be nil and similar alert behavior would be seen. The only time this is not seen is post a relocate as above.